Source/WebCore/html/HTMLSummaryElement.cpp

```diff
@@ -96,12 +98,28 @@ bool HTMLSummaryElement::isActiveSummary() const
     return details->isActiveSummary(*this);
 }
 
-static bool isClickableControl(EventTarget* target)
+static bool isInSummaryInteractiveContent(EventTarget* target)
 {
     if (!is<Element>(target))
         return false;
     auto& element = downcast<Element>(*target);
-    return is<HTMLFormControlElement>(element) || is<HTMLFormControlElement>(element.shadowHost());
+    if (is<HTMLElement>(element)) {
+        for (auto& ancestorOrSelf : lineageOfType<HTMLElement>(element)) {
+            if (is<HTMLSummaryElement>(ancestorOrSelf))
+                return false;
+            if (ancestorOrSelf.isInteractiveContent() || (is<HTMLElement>(ancestorOrSelf.shadowHost()) && downcast<HTMLElement>(ancestorOrSelf.shadowHost())->isInteractiveContent()))
+                return true;
+        }
+        return false;
+    }
+    if (is<SVGElement>(element)) {
+        for (auto& ancestorOrSelf : lineageOfType<SVGElement>(element)) {
+            if (is<SVGAElement>(ancestorOrSelf))
+                return true;
+        }
+        return false;
+    }
```

Comment:
Can we write this in a single loop?

```suggestion
    for (auto* element = &startElement; element; element = element->parentOrShadowHostNode()) {
        if (is<HTMLSummaryElement>(element))
            return false;
        if (is<HTMLElement>(element) && downcast<HTMLElement>(element).isInteractiveContent())
            return true;
        if (is<SVGAElement>(element))
            return false;
    }
```

Reply:
> Can we write this in a single loop?

Thanks much — made it so (but with a Node rather than Element, since the Element `parentOrShadowHostNode` is private).

By the way, should I use `RefPtr<ContainerNode> node` here rather than `auto* node`?